frontend/app/utils/date-utils.ts

```diff
@@ -51,3 +51,76 @@ export function getLocalizedMonths(
     text: formatter.format(Date.UTC(0, i, 1)), // formatted month name
   }));
 }
+
+/**
+ * Validates whether a given string is in the format YYYY-MM-DD.
+ *
+ * @param date - The date string to be validated.
+ * @returns Returns `true` if the string is a valid date in the format YYYY-MM-DD, otherwise `false`.
+ *
+ * @example
+ * ```javascript
+ * const validDate = "2023-05-20";
+ * const invalidDate = "20-05-2023";
+ *
+ * console.log(isValidDateString(validDate)); // true
+ * console.log(isValidDateString(invalidDate)); // false
+ * ```
+ */
+export const isValidDateString = (dateString: string) => {
+  const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
+  if (!dateRegex.test(dateString)) {
+    return false;
+  }
+  const date = new Date(dateString);
+  return !isNaN(date.getTime());
```

Comment:
Can be simplified with `date-fns` functions: [isValid](https://date-fns.org/v4.1.0/docs/isValid) and [parseISO](https://date-fns.org/v4.1.0/docs/parseISO)

```suggestion
  return isValid(parseISO(dateString));
```

Reply:
@sebastien-comeau --- I'm resolving this.